utils/results/conversion/summaryparser/summaryparser.go

```diff
@@ -221,77 +249,124 @@ func getCveIds(cves []formats.CveRow, issueId string) []string {
+	}
+	return results.ApplyHandlerToJasIssues(results.ScanResultsToRuns(iacs), sc.entitledForJas, sc.getJasHandler(jasutils.IaC, isViolationsResults))
 }
 
-func (sc *CmdResultsSummaryConverter) ParseSast(_ results.ScanTarget, sast ...*sarif.Run) (err error) {
+func (sc *CmdResultsSummaryConverter) ParseSast(_ results.ScanTarget, isViolationsResults bool, sast []results.ScanResult[[]*sarif.Run]) (err error) {
 	if !sc.entitledForJas || sc.currentScan.Vulnerabilities == nil {
 		// JAS results are only supported as vulnerabilities for now
 		return
 	}
 	if err = sc.validateBeforeParse(); err != nil {
 		return
 	}
-	if sc.currentScan.Vulnerabilities.SastResults == nil {
+	if !isViolationsResults && sc.currentScan.Vulnerabilities.SastResults == nil {
 		sc.currentScan.Vulnerabilities.SastResults = &formats.ResultSummary{}
 	}
-	return results.PrepareJasIssues(sast, sc.entitledForJas, sc.getJasHandler(jasutils.Sast))
+	if isViolationsResults {
+		if sc.currentScan.Violations.SastResults == nil {
+			sc.currentScan.Violations.SastResults = &formats.ResultSummary{}
+		}
+		sc.currentScan.Violations.Watches = utils.UniqueUnion(sc.currentScan.Violations.Watches, getJasScansWatches(sast...)...)
+	}
+	return results.ApplyHandlerToJasIssues(results.ScanResultsToRuns(sast), sc.entitledForJas, sc.getJasHandler(jasutils.Sast, isViolationsResults))
 }
 
-func (sc *CmdResultsSummaryConverter) getJasHandler(scanType jasutils.JasScanType) results.ParseJasFunc {
+func (sc *CmdResultsSummaryConverter) getJasHandler(scanType jasutils.JasScanType, violations bool) results.ParseJasFunc {
 	return func(run *sarif.Run, rule *sarif.ReportingDescriptor, severity severityutils.Severity, result *sarif.Result, location *sarif.Location) (err error) {
 		if location == nil {
 			// Only count the issue if it has a location
 			return
 		}
-		// Get the scanType count
+		// Get the scanType count and status
+		resultStatus := formats.NoStatus
 		var count *formats.ResultSummary
 		switch scanType {
 		case jasutils.Secrets:
-			count = sc.currentScan.Vulnerabilities.SecretsResults
+			if tokenStatus := results.GetResultPropertyTokenValidation(result); tokenStatus != "" {
+				resultStatus = tokenStatus
+			}
+			if violations {
+				count = sc.currentScan.Violations.SecretsResults
+			} else {
+				count = sc.currentScan.Vulnerabilities.SecretsResults
+			}
 		case jasutils.IaC:
-			count = sc.currentScan.Vulnerabilities.IacResults
+			if violations {
+				count = sc.currentScan.Violations.IacResults
+			} else {
+				count = sc.currentScan.Vulnerabilities.IacResults
+			}
 		case jasutils.Sast:
-			count = sc.currentScan.Vulnerabilities.SastResults
+			if violations {
+				count = sc.currentScan.Violations.SastResults
+			} else {
+				count = sc.currentScan.Vulnerabilities.SastResults
+			}
 		}
 		if count == nil {
 			return
 		}
-		// PrepareJasIssues calls the handler for each issue (location)
+		// Calls the handler for each issue (location)
```

Comment:
I dont understand what exactly we are calling here. we are only setting the map with values and returning this handler function

Reply:
copy/paste leftovers that were not removed.